### PR TITLE
Enabled C++20 (main)

### PR DESCRIPTION
**Pull request by korydraughn**
Requires latest versions of externals packages. Specifically, fmt v8.1.1 due to https://github.com/fmtlib/fmt/commit/023c2018f78a66c9acb3b9c7522ec431a210eb8c.

### CODE REVIEW

File: server/core/src/irods_server_control_plane.cpp
Comment (korydraughn) [inline server/core/src/irods_server_control_plane.cpp:146]: clean up

**Comment by korydraughn** [comment]
Cleaned up commits.

**Review by alanking — APPROVED** [review]

File: plugins/api/src/atomic_apply_metadata_operations.cpp
Comment (korydraughn) [reply plugins/api/src/atomic_apply_metadata_operations.cpp:181]: Yes. Will update.

File: server/core/src/catalog_utilities.cpp
Comment (korydraughn) [reply server/core/src/catalog_utilities.cpp:124]: Will update

File: plugins/api/src/atomic_apply_metadata_operations.cpp
Comment (korydraughn) [reply plugins/api/src/atomic_apply_metadata_operations.cpp:182]: done

File: server/core/src/catalog_utilities.cpp
Comment (korydraughn) [reply server/core/src/catalog_utilities.cpp:127]: done

File: CMakeLists.txt
Comment (korydraughn) [inline CMakeLists.txt:38]: should we disable compiler extensions?

what do we gain/lose by doing this?

File: CMakeLists.txt
Comment (korydraughn) [inline CMakeLists.txt:151]: this should be captured in an issue.

once boost is able to detect `std::invoke_result`, this macros should be removed.

File: unit_tests/CMakeLists.txt
Comment (korydraughn) [inline unit_tests/CMakeLists.txt:33]: see previous comment about `BOOST_ASIO_HAS_STD_INVOKE_RESULT`

File: CMakeLists.txt
Comment (korydraughn) [reply CMakeLists.txt:38]: @alanking @SwooshyCueb any thoughts on this?

**Comment by korydraughn** [comment]
@trel this PR requires that the latest versions of externals packages be published to core-dev and unstable.

right now, no one can build this unless they build their own copy of the externals packages.

**Comment by trel** [comment]
can you confirm only fmt8.1.1 is missing at the moment?

**Comment by korydraughn** [comment]
checking ...

File: CMakeLists.txt
Comment (alanking) [reply CMakeLists.txt:38]: Disabling this feels like the right approach until we find that we need some extension.

The documentation states...

> Note that because most compilers enable extensions by default, this may expose portability bugs in user code or in the headers of third-party dependencies.

Found here: https://cmake.org/cmake/help/latest/manual/cmake-compile-features.7.html#availability-of-compiler-extensions

But I'm probably not the most trustworthy authority for that question :)

**Comment by trel** [comment]
on it

File: CMakeLists.txt
Comment (korydraughn) [reply CMakeLists.txt:38]: i had the same thoughts around that.

unless @SwooshyCueb has any thoughts on it, i'm leaning towards disabling extensions. if we need them, we can re-enable them.

**Comment by korydraughn** [comment]
waiting for build to succeed.

**Comment by korydraughn** [comment]
builds succeeded.

added #'s

File: CMakeLists.txt
Comment (korydraughn) [reply CMakeLists.txt:38]: marking as resolved.

if we need extensions, we can re-enable them in a later commit.

**Comment by trel** [comment]
before we merge/close this... did you capture your comments above as new issues?

**Comment by korydraughn** [comment]
I was waiting until this is merged so I can reference the same code from the main branch. That is also the reason I haven't marked those as resolved.

I didn't want to link to code in the PR.

File: unit_tests/CMakeLists.txt
Comment (korydraughn) [reply unit_tests/CMakeLists.txt:33]: captured in issue.

File: CMakeLists.txt
Comment (SwooshyCueb) [reply CMakeLists.txt:38]: Apologies, this didin't show up in my notifications for some reason.

I agree with disabling extensions if we don't need them or have a specific reason to keep them enabled.
This isn't something I would typically do, myself, but I don't fault your logic here and I ultimately think it's the right choice for the project.

**Pull request merged**
